File: app/src/main/java/com/absinthe/anywhere_/view/ObservableTimePickerDialog.java
package com.absinthe.anywhere_.view;

import android.app.TimePickerDialog;
import android.content.Context;

public class ObservableTimePickerDialog extends TimePickerDialog {

    private OnCancelledListener mListener;

    public ObservableTimePickerDialog(Context context, OnTimeSetListener listener, OnCancelledListener cancelListener, int hourOfDay, int minute, boolean is24HourView) {
        super(context, listener, hourOfDay, minute, is24HourView);
        mListener = cancelListener;
    }

    @Override
    public void cancel() {
        super.cancel();
        mListener.onCancel();
    }

    public interface OnCancelledListener {
        void onCancel();
    }
}

File: app/src/main/java/com/absinthe/anywhere_/view/settings/ObservableTimePickerDialog.kt
package com.absinthe.anywhere_.view.settings

import android.app.TimePickerDialog
import android.content.Context

class ObservableTimePickerDialog(
        context: Context?,
        listener: OnTimeSetListener?,
        private val mListener: OnCancelledListener,
        hourOfDay: Int, minute: Int, is24HourView: Boolean)
    : TimePickerDialog(context, listener, hourOfDay, minute, is24HourView) {

    override fun cancel() {
        super.cancel()
        mListener.onCancel()
    }

    interface OnCancelledListener {
        fun onCancel()
    }

}